src/ocean/text/Util.d

```diff
@@ -785,6 +788,136 @@ bool isSpace(T) (T c)
            return (c <= 32 && (c is ' ' || c is '\t' || c is '\r' || c is '\n' || c is '\f' || c is '\v')) || (c is '\u2028' || c is '\u2029');
 }
 
+/*******************************************************************************
+
+    Enumeration of the different ways in which a string can be padded in order
+    to increase its length.
+
+*******************************************************************************/
+
+public enum PadType : ubyte
+{
+    PadPre,    // Padding is only applied before the string
+    PadPost,   // Padding is only applied after the string
+    PadAround  // Padding is applied on both sides of the string
+}
+
+/*******************************************************************************
+
+    Shortens/Lengthens a string as necessary to get it to a desired length.
+
+    Note 1:
+        If the string is shortened, an ellipsis (3 dots) will be appended to
+        indicate intentional truncation.
+
+    Note 2:
+        The length specified must be at least 4. If this constraint were absent
```

Comment:
Adding the `...` is [a tango thing](https://github.com/sociomantic-tsunami/ocean/blob/v2.x.x/src/ocean/text/convert/Layout_tango.d#L1138-L1141). We just use the existing chopping functionality here.